src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -0,0 +1,134 @@
+        public Http3ControlStream EncoderStream { get; set; }
+        public Http3ControlStream DecoderStream { get; set; }
+
+        private readonly ConcurrentDictionary<int, Http3Stream> _streams = new ConcurrentDictionary<int, Http3Stream>();
+
+        public Http3Connection(HttpConnectionContext context)
+        {
+            Context = context;
+            DynamicTable = new DynamicTable(0);
+            // TODO qpack decoders need to be one per stream?
+
+        }
+
+        public async Task ProcessRequestsAsync<TContext>(IHttpApplication<TContext> application)
+        {
+            var streamFeature = Context.ConnectionFeatures.Get<IQuicStreamListenerFeature>();
+            if (streamFeature == null)
+            {
+                throw new Http3ConnectionException();
+            }
+
+            try
+            {
+                var i = 0;
+                while (true)
+                {
+                    var connectionContext = await streamFeature.AcceptAsync();
+                    if (connectionContext == null)
+                    {
+                        break;
+                    }
+
+                    // Need to keep calling AcceptAsync here.
+                    var httpConnectionContext = new HttpConnectionContext
+                    {
+                        ConnectionId = connectionContext.ConnectionId + i, // TODO this is just adding an int for the stream id, which isn't great.
+                        ConnectionContext = connectionContext,
+                        Protocols = Context.Protocols,
+                        ServiceContext = Context.ServiceContext,
+                        ConnectionFeatures = connectionContext.Features,
+                        MemoryPool = Context.MemoryPool,
+                        Transport = connectionContext.Transport,
+                        TimeoutControl = Context.TimeoutControl,
+                        LocalEndPoint = connectionContext.LocalEndPoint as IPEndPoint,
+                        RemoteEndPoint = connectionContext.RemoteEndPoint as IPEndPoint
+                    };
+
+                    IHttp3Stream stream;
+                    if (httpConnectionContext.ConnectionFeatures.Get<IUnidirectionalStreamFeature>() != null)
+                    {
+                        stream = new Http3ControlStream(this, httpConnectionContext);
+                    }
+                    else
+                    {
+                        var http3Stream = new Http3Stream(this, httpConnectionContext);
+                        stream = http3Stream;
+                        _streams[i] = http3Stream;
+                        // TODO Figure out how to make ids more unique
+                    }
+
+                    i++;
+                    _ = Task.Run(() => stream.ProcessRequestAsync(application));
+                }
+            }
+            finally
+            {
+                foreach (var stream in _streams.Values)
```

Comment:
This needs to be done gracefully where we abort the streams and drain the tasks.

Reply:
Yes. What should I do here for now? Not drain at all?

Reply:
Yea, we'll file an issue.